metaseq/models/transformer_decoder.py

```diff
@@ -119,7 +119,8 @@ def __init__(self, args, dictionary, embed_tokens):
             if args.decoder_learned_pos and not self.use_alibi
             else None
         )
-        self.embed_positions.to(device).to(dtype)
+        if self.embed_positions is not None:
```

Comment:
We had cases where `self.embed_positions` were None at this point?

Reply:
Yeah -L119 is the condition for no embed_positions - happens for `transformer_lm_gpt2_tiny` at least. But the `decoder_learned_pos` option defaults to False, so presumably this is usually overridden by user args?

Reply:
Oof... yeah... let's do the not-so-great thing and change that default to `True`.  We clobber this with our sweep scripts all the time so might as well bite the bullet here and make it consistent throughout (opened https://github.com/facebookresearch/metaseq/issues/435 to track; good work flagging all of these papercuts 😅  )